examples/vision/keypoint_detection.py

```diff
@@ -0,0 +1,441 @@
+            if new_image.shape[-1] == 4:
+                new_image = Image.fromarray(new_image)
+                new_image = np.array(new_image.convert("RGB"))
+            batch_images[i,] = new_image
+
+            # Parse the coordinates from the new keypoint object.
+            kp_temp = []
+            for keypoint in new_kps_obj:
+                kp_temp.append(np.nan_to_num(keypoint.x))
+                kp_temp.append(np.nan_to_num(keypoint.y))
+
+            # More on why this reshaping later.
+            batch_keypoints[i,] = np.array(kp_temp).reshape(1, 1, 24 * 2)
+
+        # Scale the coordinates to [0, 1] range.
+        batch_keypoints = batch_keypoints / IMG_SIZE
+
+        return (batch_images, batch_keypoints)
+
+
+"""
+To know more about how to operate with keypoints in `imgaug` check out [this document](https://imgaug.readthedocs.io/en/latest/source/examples_keypoints.html). 
+"""
+
+"""
+## Define augmentation transforms
+"""
+
+train_aug = iaa.Sequential(
+    [
+        iaa.Resize(IMG_SIZE, interpolation="linear"),
+        iaa.Fliplr(0.3),
+        # `Sometimes()` applies a function randomly to the inputs with
+        # a given probability (0.3, in this case).
+        iaa.Sometimes(0.3, iaa.Affine(rotate=10, scale=(0.5, 0.7))),
+    ]
+)
+
+test_aug = iaa.Sequential([iaa.Resize(IMG_SIZE, interpolation="linear")])
+
+"""
+## Create training and validation splits
+"""
+
+np.random.shuffle(samples)
+train_keys, validation_keys = (
+    samples[int(len(samples) * 0.15) :],
+    samples[: int(len(samples) * 0.15)],
+)
+
+
+"""
+## Data generator investigation
+"""
+
+train_dataset = KeyPointsDataset(train_keys, train_aug)
+validation_dataset = KeyPointsDataset(validation_keys, test_aug, train=False)
+
+print(f"Total batches in training set: {len(train_dataset)}")
+print(f"Total batches in validation set: {len(validation_dataset)}")
+
+sample_images, sample_keypoints = next(iter(train_dataset))
+assert sample_keypoints.max() == 1.0
+assert sample_keypoints.min() == 0.0
+
+sample_keypoints = sample_keypoints[:4].reshape(-1, 24, 2) * IMG_SIZE
+visualize_keypoints(sample_images[:4], sample_keypoints)
+
+"""
+## Model building
+
+The [Stanford dogs dataset](http://vision.stanford.edu/aditya86/ImageNetDogs/) (on which
+the StanfordExtra dataset is based) was built using the [ImageNet-1k dataset](http://image-net.org/).
+So, it is likely that the models pre-trained on the ImageNet-1k dataset would be useful
+for this task. We will use a MobileNetV2 pre-trained on this dataset as a backbone to
+extract meaningful features from the images and then pass those to a custom regression
+head for predicting coordinates. 
+"""
+
+
+def get_model():
+    backbone = keras.applications.MobileNetV2(
+        weights="imagenet", include_top=False, input_shape=(IMG_SIZE, IMG_SIZE, 3)
+    )
+    inputs = layers.Input((IMG_SIZE, IMG_SIZE, 3))
+    x = keras.applications.mobilenet_v2.preprocess_input(inputs)
+    x = backbone(x, training=False)
```

Comment:
This would run the batch norm layers in inference mode, we think we should specify why this is being done

Reply:
I didn't specify it for brevity. I can add a sentence to refer to the fine-tuning guide where it's been explained.

Reply:
If the intent is to keep `backbone` frozen, we should do `backbone.trainable = False` (which will also run the BN layers in inference mode). An issue here is that we're adding randomly initialized layers on top of pretrained features, which largely negates the usefulness of the pretrained features.

Reply:
I went with `backbone.trainable = False` and added comments. 

> An issue here is that we're adding randomly initialized layers on top of pretrained features, which largely negates the usefulness of the pretrained features.

So, how would you suggest we handle this? In transfer learning settings, appending randomly initialized layer(s) on top of pre-trained layers is common.